internal/resource/config/pingoneconnection/ping_one_connection_resource.go

```diff
@@ -0,0 +1,311 @@
+	_ resource.ResourceWithConfigure   = &pingOneConnectionResource{}
+	_ resource.ResourceWithImportState = &pingOneConnectionResource{}
+)
+
+// PingOneConnectionResource is a helper function to simplify the provider implementation.
+func PingOneConnectionResource() resource.Resource {
+	return &pingOneConnectionResource{}
+}
+
+// pingOneConnectionResource is the resource implementation.
+type pingOneConnectionResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type pingOneConnectionResourceModel struct {
+	Id                               types.String `tfsdk:"id"`
+	ConnectionId                     types.String `tfsdk:"connection_id"`
+	Name                             types.String `tfsdk:"name"`
+	Description                      types.String `tfsdk:"description"`
+	Active                           types.Bool   `tfsdk:"active"`
+	Credential                       types.String `tfsdk:"credential"`
+	EncryptedCredential              types.String `tfsdk:"encrypted_credential"`
+	CredentialId                     types.String `tfsdk:"credential_id"`
+	PingOneConnectionId              types.String `tfsdk:"ping_one_connection_id"`
+	EnvironmentId                    types.String `tfsdk:"environment_id"`
+	CreationDate                     types.String `tfsdk:"creation_date"`
+	OrganizationName                 types.String `tfsdk:"organization_name"`
+	Region                           types.String `tfsdk:"region"`
+	PingOneManagementApiEndpoint     types.String `tfsdk:"ping_one_management_api_endpoint"`
+	PingOneAuthenticationApiEndpoint types.String `tfsdk:"ping_one_authentication_api_endpoint"`
+}
+
+// GetSchema defines the schema for the resource.
+func (r *pingOneConnectionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages Ping One Connection",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Description: "The name of the PingOne Connection",
+				Required:    true,
+			},
+			"description": schema.StringAttribute{
+				Description: "The description of the PingOne Connection",
+				Optional:    true,
+			},
+			"active": schema.BoolAttribute{
+				Description: "Whether the PingOne Connection is active. Defaults to true.",
+				Computed:    true,
+				Optional:    true,
+				Default:     booldefault.StaticBool(true),
+			},
+			"credential": schema.StringAttribute{
+				Description: "The credential for the PingOne connection.",
+				Optional:    true,
+				Sensitive:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"encrypted_credential": schema.StringAttribute{
+				Description: "The encrypted credential for the PingOne connection.",
+				Sensitive:   true,
+				Optional:    false,
+				Computed:    true,
+			},
+			"credential_id": schema.StringAttribute{
```

Comment:
Do we want all of these in the resource? I know some other times we've left some of the read only fields for the data source. Not sure what is best.

Reply:
we left these in to match the API

Reply:
Ok, I had thought we had removed these in some other resources (key pairs) but it looks like we put them back in.